cfme/cloud/instance/__init__.py

```diff
@@ -100,25 +95,60 @@ def __init__(self, name, provider, template_name=None, appliance=None):
         super(Instance, self).__init__(name=name, provider=provider, template_name=template_name)
         Navigatable.__init__(self, appliance=appliance)
 
-    def create(self):
+    def create(self, form_values, cancel=False):
         """Provisions an instance with the given properties through CFME
+
+        Args:
+            form_values: dictionary of form values for provisioning, structured into tabs
+
+        Note:
+            Calling create on a sub-class of instance will generate the properly formatted
+            dictionary when the correct fields are supplied.
         """
-        raise NotImplementedError('create is not implemented.')
-
-    def load_details(self, refresh=False):
-        navigate_to(self, 'Details')
-        # BZ 1389299
-        # TODO: add refresh support back
-
-    def get_detail(self, *args, **kwargs):
-        # TODO: remove when bug 1389299 is fixed to use common.vm.get_detail()
-        # Navigate to all first to force reload of details screen without using refresh
-        navigate_to(self, 'All')
-        self.load_details()
-        if kwargs.get('icon_href', False):
-            return InfoBlock.icon_href(*kwargs.get('properties'))
+        view = navigate_to(self, 'Provision')
+
+        # Only support 1 security group for now
+        # TODO: handle multiple
+        if 'environment' in form_values and 'security_groups' in form_values['environment'] and \
+                isinstance(form_values['environment']['security_groups'], (list, tuple)):
+
+            first_group = form_values['environment']['security_groups'][0]
+            recursive_update(form_values, {'environment': {'security_groups': first_group}})
+
+        view.form.fill(form_values)
+
+        if cancel:
+            view.form.cancel_button.click()
+            # Redirects to Instance All, but flash is defined in BaseLoggedInPage so use directly
+            view = self.browser.create_view(BaseLoggedInPage)
+            wait_for(func=view.flash.assert_success_message, func_arg=self.PROVISION_CANCEL,
+                     fail_condition=[], timeout=10, delay=2, message='wait for flash success')
```

Comment:
There is no such arg as func_arg, you probably wanted to pass a list to func_args.

Flash messages widget does not return anything, always returns None then.